jerry-core/api/jerry.c

```diff
@@ -3458,14 +3457,19 @@ jerry_create_arraybuffer_external (const jerry_length_t size, /**< size of the b
   jerry_assert_api_available ();
 
 #if ENABLED (JERRY_ES2015_BUILTIN_TYPEDARRAY)
-  if (size == 0 || buffer_p == NULL)
+  ecma_object_t *arraybuffer;
+
+  if (JERRY_UNLIKELY (size == 0 || buffer_p == NULL))
+  {
+    arraybuffer = ecma_arraybuffer_new_object_external (0, NULL, NULL);
```

Comment:
Is it ok to not register the free callback?

Reply:
You are right. It's good to know when the ArrayBuffer is garbage collected.